src/org/zaproxy/zap/extension/ascanrulesBeta/BackupFileDisclosure.java

```diff
@@ -497,6 +497,12 @@ private void findBackupFile(HttpMessage originalMessage) throws Exception {
 				sendAndReceive(requestmsg, false);
 				disclosedData = requestmsg.getResponseBody().getBytes();
 				int requestStatusCode = requestmsg.getResponseHeader().getStatusCode();
+				
+				if (!gives404s && getBaseMsg().getResponseHeader().getStatusCode() != requestStatusCode) {
```

Comment:
Looks good but could be done in the following if statement (replacing the second `requestStatusCode != HttpStatus.SC_NOT_FOUND`, line 511).

Reply:
(With `getBaseMsg().getResponseHeader().getStatusCode() == requestStatusCode`)

Reply:
No problem.